src/services/data-query.js

```diff
@@ -81,48 +112,115 @@ function QueryService(){
    * @returns {Promise<unknown>}
    */
   this.bbox = function({ bbox, feature_count=this.project.getQueryFeatureCount(), filterConfig={}, multilayers=false, condition=this.condition, layersFilterObject = {SELECTEDORALL: true, FILTERABLE: true, VISIBLE: true}}={}) {
-    const layers = getMapLayersByFilter(layersFilterObject, condition);
-    const request = getQueryLayersPromisesByBBOX(layers, {
-      bbox,
-      feature_count,
-      filterConfig,
-      multilayers,
-    });
-    return this.handleRequest(request, {
+    const hasExternalLayersSelected = this.hasExternalLayerSelected({ type: "vector" });
+    /**
+     * @since v3.8
+     * @type {{external: {add: boolean, filter: {SELECTED: boolean}}, bbox, type: string, filterConfig}}
+     */
+    const query = {
       bbox,
       type: 'bbox',
-    });
+      filterConfig,
+      external: {
+        add: true,
+        filter: {
+          SELECTED: externalSelectedLayers.length > 0
```

Comment:
The `externalSelectedLayers` array could be undefined?

It seems to be undeclared within the `bbox()` scope.

Reply:
@Raruto solved https://github.com/g3w-suite/g3w-client/pull/356/commits/03c52cc14f2f8290ff54f0372e47ce87798776d8